src/librustc_mir/transform/const_prop.rs

```diff
@@ -988,8 +1023,61 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
             | TerminatorKind::GeneratorDrop
             | TerminatorKind::FalseEdges { .. }
             | TerminatorKind::FalseUnwind { .. } => {}
-            //FIXME(wesleywiser) Call does have Operands that could be const-propagated
-            TerminatorKind::Call { .. } => {}
+            TerminatorKind::Call { ref mut args, .. } => {
+                // FIXME(felix91gr) Finish this
+                for opr in args {
+                    // If it isn't a place, then it's a constant,
+                    // and thus we don't need to propagate towards it.
+                    trace!("Checking if arg {:?} is not a constant", &opr);
+                    if let Some(place) = opr.place() {
```

Comment:
I think simplifying this by using some of the methods we already have might help narrow down the issue. Pretty much all of this from this line to 1041 could be replaced by using `self.eval_operand(Operand)`.

Reply:
Oh! That's true! 

That would also evaluate `Operand::Constant` (currently the code is discarding all `Operand`s that aren't `Operand::Move` or `Operand::Copy`). I'm guessing that evaluating constants is innocuous in this context, right? As in, it shouldn't introduce anything weird. If an `Operand` is of the `Constant` variant, then that means that it's like a `const` or a literal, right? That sort of thing?

Reply:
Yeah, it's fine. Evaluating a `Constant` will just give us a copy of that constant.

Reply:
Oh, okay, that's awesome. Maybe it seemed obvious, but I had to ask <3